### PR TITLE
[Bug] [flink-connector-jdbc] change jdbc related dependency scope from provided to default scope

### LINKED ISSUE #1463: [Bug] [flink-connector-jdbc] jdbc related dependency scope is provided
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

location on `seatunnel/pom.xml`, `flink-connector-jdbc_2.11` scope is `provided` and seatunnel release doesn't contain flink-connector-jdbc dependency.
```
<flink.scope>provided</flink.scope>

......

<dependency>
    <groupId>org.apache.flink</groupId>
    <artifactId>flink-connector-jdbc_${scala.binary.version}</artifactId>
    <version>${flink.version}</version>
    <scope>${flink.scope}</scope>
</dependency>
```
The same problem is MySQL JDBC driver dependency, and for license compliance, we have to warn people adding driver by themselves.
```
<!--Because the license is not in compliance, if you need to use MySQL, you can add it yourself-->
<dependency>
    <groupId>mysql</groupId>
    <artifactId>mysql-connector-java</artifactId>
    <version>${mysql.version}</version>
    <scope>test</scope>
</dependency>
```

### SeaTunnel Version

dev

### SeaTunnel Config

```conf
not needed
```

### Running Command

```shell
not needed
```

### Error Exception

```log
not needed
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kalencaya**
## Purpose of this pull request
Fix #1463 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: docs/en/flink/configuration/sink-plugins/Jdbc.md [hunk 1]
```diff
@@ -22,7 +22,9 @@ Write data through jdbc
 
 ### driver [string]
 
-Driver name, such as `com.mysql.jdbc.Driver`
+Driver name, such as `com.mysql.cj.jdbc.Driver`.
+
+Warn: for license compliance, you have to provide MySQL JDBC driver yourself.
```
Comment (ruanwenjun) [inline docs/en/flink/configuration/sink-plugins/Jdbc.md]: It might be better to describe how to add `mysql-connector-java`.

Comment (kalencaya) [reply docs/en/flink/configuration/sink-plugins/Jdbc.md]: Good advise, I'll add more guiding info.

**Review by ruanwenjun — DISMISSED** [review]
LGTM

**Comment by kalencaya** [comment]
more PR intended for removing never used and outdated dependency for `flink-jdbc_${scala.binary.version}`, which is renamed to `flink-connector-jdbc_${scala.binary.version}` from Flink `1.11.0` version.

**Review by CalvinKirs — APPROVED** [review]
Well done

**Pull request merged**